src/gr11review/part2/Utility.java

```diff
@@ -1,5 +1,432 @@
 package gr11review.part2;
 
+// Ethan's Imports
+import java.io.*;
+import java.util.*;
+
+// Nick's Imports
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+
 public class Utility {
+
+/* 
+* Given a string, return the sum of the numbers appearing in the string, ignoring all other characters. A number is a series of 1 or more digit chars in a row. 
+* @param str = User input string that will be used for the method
+* @return intResult = the sum of the numbers in the String "str"
+* @author E.Lai
+*/
+  // Methods 2
+  public static int sumNumbers(String str){
+
+    // Variables
+    int intLoop;
+    int intLoop2;
+    int intResult;
+    char chrCharacters;
+    String strNumbers;
+
+    // Initializing Variables
+    strNumbers = "";
+    intResult = 0;
+
+    // For loop that allows the program to count how many characters are in the string
+    for (intLoop = 0; intLoop < str.length(); intLoop++) {
+      chrCharacters = str.charAt(intLoop);
+
+      // If the next character the program counts is a number, adds the number character to a string
+      if (Character.isDigit(chrCharacters)) {
+        strNumbers = strNumbers + chrCharacters;
+
+      // If the next character the program counts is not a number, adds the string stored with numbers into intResult. Then the program resets the string.
+      } else if (!strNumbers.equals("")) {
+        intResult = intResult + Integer.parseInt(strNumbers);
+        strNumbers = "";
+      }
+    }
+    
+    // If strNumbers still has numbers stored in it by the end of the string, it adds the remaining numbers into intResult.
+    if (!strNumbers.equals("")) {
+      intResult = intResult + Integer.parseInt(strNumbers);
+    }
+    return intResult;
+  }
+
+/* 
+* Write a method alphaWord(String filenametxt) such that given the name of a file filenametxt that contains a single word on each line,returns the word that is alphabetically first. 
+* @param filenametxt = Name of the text file that will be analyzed
+* @return strResultingWord = Alphabetically sorted word
+* @author E.Lai
+*/
+  // File IO - Read 2 Method
+  public static String alphaWord(String filenametxt) throws IOException {
+    BufferedReader TextFile = new BufferedReader(new FileReader(filenametxt));
+
+    // Variables
+    String strTempWord;
+    String strResultingWord;
+    int intAlphaCompare;
+    
+    // Initializing Variables
+    strTempWord = TextFile.readLine();
+    strResultingWord = TextFile.readLine();
+
+    // While Loop that keeps on looping strTempWord until it becomes null (end of file)
+    while (strTempWord != null) {
+
+      // Compares the current word (strTempWord) to the first word of the file (strResultingWord)
+			intAlphaCompare = strResultingWord.compareToIgnoreCase(strTempWord);
```

Comment:
@EthanLai8 Method fails when the test file consists of only one word.

```
java.lang.NullPointerException: Cannot invoke "String.compareToIgnoreCase(String)" because "strResultingWord" is null
	at gr11review.part2.Utility.alphaWord(Utility.java:81)
```